sw/device/silicon_creator/lib/sigverify/sphincsplus/params.h

```diff
@@ -0,0 +1,107 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Derived from code in the SPHINCS+ reference implementation (CC0 license):
+// https://github.com/sphincs/sphincsplus/blob/ed15dd78658f63288c7492c00260d86154b84637/ref/params/params-sphincs-shake-128s.h
+
+/**
+ * This file represents the SPHINCS+ parameter set shake-128s, meaning:
+ * - The hash function is SHAKE-256
+ * - >= 128 bits of security for up to 2^64 signatures
+ * - The parameter set is optimized to be small "s" rather than fast "f"
+ *   - The "fast" variant is faster for signing but actually slower for
+ *     verification, so "s" is a better fit given we don't care about signing
+ *     speed.
+ *
+ * For more details on parameterization, see the SPHINCS+ paper:
+ * https://sphincs.org/data/sphincs+-paper.pdf
+ */
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_SIGVERIFY_SPHINCSPLUS_PARAMS_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_SIGVERIFY_SPHINCSPLUS_PARAMS_H_
+
+/* Hash output length in bytes. */
+#define SPX_N 16
```

Comment:
I think we should either vendor this file, or (my preference) rewrite the constants here in line with our style guide:
- Replace all `define`s with an anonymous `enum` -- assuming these are _published_ constants that don't have any other type-significance.
- Did we settle on a parameter set or do we want to keep all of these sets + tests + tooling for them? It would be great if we could replace all `ifdef`s with a single branch.

Reply:
I think it's reasonable to just refactor with an anonymous `enum`. Vendoring would be slightly complicated because this file is more than just a copy; it also includes a few additional derived constants for sizes in 32b words.